ofrak_core/ofrak/core/ubifs.py

```diff
@@ -0,0 +1,176 @@
+import subprocess
+import tempfile
+from dataclasses import dataclass
+import logging
+
+from ofrak import Identifier, Analyzer
+from ofrak.component.packer import Packer
+from ofrak.component.unpacker import Unpacker
+from ofrak.resource import Resource
+from ofrak.core.filesystem import File, Folder, FilesystemRoot, SpecialFileType
+from ofrak.core.binary import GenericBinary
+
+from ofrak.model.component_model import ComponentExternalTool
+from ofrak_type.range import Range
+
+from ubireader.ubifs.defines import UBIFS_NODE_MAGIC
+
+from ubireader import ubi_io
+from ubireader.ubifs import ubifs as ubireader_ubifs
+from ubireader.ubifs.defines import PRINT_UBIFS_KEY_HASH, PRINT_UBIFS_COMPR
+from ubireader.utils import guess_leb_size
+
+LOGGER = logging.getLogger(__name__)
+
+MKFS_UBIFS_TOOL = ComponentExternalTool(
+    "mkfs.ubifs",
+    "http://www.linux-mtd.infradead.org/faq/ubifs.html",
+    install_check_arg="--help",
+    apt_package="mtd-utils",
+    brew_package="",  # This isn't compatible with macos, but there may be an alternative tool to do the bidding.
+)
+
+
+@dataclass
+class SuperblockNode:
+    max_leb_count: int
+    default_compr: str  # PRINT_UBIFS_COMPR
+    fanout: int
+    key_hash: str  # PRINT_UBIFS_KEY_HASH
+    orph_lebs: int
+    log_lebs: int
+
+
+@dataclass
+class Ubifs(GenericBinary, FilesystemRoot):
+    """
+    http://www.linux-mtd.infradead.org/doc/ubifs.html
+    Minimum number of bytes per transaction (see http://www.linux-mtd.infradead.org/doc/ubi.html#L_min_io_unit)
+    Size of logical erase blocks and superblock parameters required for repacking.
```

Comment:
Can you make this docstring a declarative sentence about what this is? https://github.com/redballoonsecurity/ofrak/blob/master/ofrak_core/ofrak/core/uimage.py#L193 is a good example.

Also, if you want to describe the dataclasses attributes, you should use the following notation:

```
:ivar min_io_size: minimum number of bytes per transaction (see http://www.linux-mtd.infradead.org/doc/ubi.html#L_min_io_unit)
:ivar leb_size: size of logical erase blocks and superblock parameters required for repacking.
:ivar superblock: A SuberblockNode
``` 

Both of this make the auto-generated docs easier to read, for example: https://ofrak.com/docs/reference/ofrak/core/uimage.html